### PR TITLE
Add `--function` to `set` completion

### PR DESCRIPTION
**Pull request by kidonng**
## Description

Seems like an oversight in #8145.

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: share/completions/set.fish [hunk 1]
```diff
@@ -71,6 +71,7 @@ end
 complete -c set -n "fish_is_nth_token 1" -s e -l erase -d "Erase variable"
 complete -c set -n "fish_is_nth_token 1" -s x -l export -d "Export variable to subprocess"
 complete -c set -n "fish_is_nth_token 1" -s u -l unexport -d "Do not export variable to subprocess"
+complete -c set -n "fish_is_nth_token 1" -s f -l function -d "Make variable scope function"
```
Comment (faho) [inline share/completions/set.fish:74]: "Make variable function-scoped"

Comment (kidonng) [reply share/completions/set.fish:74]: Agree, I'm just following the existing examples. May as well change other descriptions to use same format.

Comment (faho) [reply share/completions/set.fish]: The other examples are adjectives. A variable "is local", a variable "is global".

A variable is not "function".

So the others can stay.

Comment (kidonng) [reply share/completions/set.fish]: Seems like we should not just say "make variable scope universal" so there will be inconsistency between descriptions. That's fine, I reverted other changes.

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**